File: api/marvel.py
from api.base import Base
from api.modules.characters import Characters
from api.modules.events import Events
from api.modules.series import Series
from api.modules.comics import Comics
from api.modules.creators import Creators
from api.modules.stories import Stories


class Marvel(Base):
    def __init__(self, PUBLIC_KEY, PRIVATE_KEY):
        """
        Marvel main class to access different modules and their methods.
        :param PUBLIC_KEY: str
        :param PRIVATE_KEY: str
        """
        super().__init__(PUBLIC_KEY, PRIVATE_KEY)
        self.characters = Characters(requester=self.requester)
        self.events = Events(requester=self.requester)
        self.series = Series(requester=self.requester)
        self.comics = Comics(requester=self.requester)
        self.creators = Creators(requester=self.requester)
        self.stories = Stories(requester=self.requester)

File: api/marvel/marvel.ts
import Base from "./base.js"
import Characters from "./modules/characters.js";
import Comics from "./modules/comics.js";
import Creators from "./modules/creators.js";
import Events from "./modules/events.js";
import Series from "./modules/series.js";
import Stories from "./modules/stories.js";

export default class marvel extends Base {
    
    characters: Characters;
    comics: Comics;
    creators: Creators;
    events: Events;
    series: Series;
    stories: Stories;

    constructor(PUBLIC_KEY: string, PRIVATE_KEY: string){
        
        super(PUBLIC_KEY, PRIVATE_KEY);
        this.characters = new Characters(this.Requester);
        this.comics = new Comics(this.Requester);
        this.creators = new Creators(this.Requester);
        this.events = new Events(this.Requester);
        this.series = new Series(this.Requester);
        this.stories = new Stories(this.Requester);
    }




}